plugins/presentation/src/components/internal/Table/TableControls.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+  import { createEventDispatcher, getContext } from 'svelte'
+  import { filter } from './utils'
+  const dispatch = createEventDispatcher()
+  const stateContext = getContext('table-state')
+
+  export let index = -1
+  export let text = ''
+
+  export let labels = {
+    placeholder: 'Search'
+  }
+
+  function onSearch (event) {
+    const state = stateContext.getState()
```

Comment:
I've not catch controls are inside Table? Why not pass state in this case, or it should control component from out side, please Check how our markdown reference editor control are implemented.

Reply:
yeah, I'm gonna use writable store instead of context to keep visible rows up to date.